httpx/_types.py

```diff
@@ -38,7 +40,9 @@
 ]
 
 HeaderTypes = Union[
-    "Headers", Dict[StrOrBytes, StrOrBytes], Sequence[Tuple[StrOrBytes, StrOrBytes]],
+    "Headers",
+    MutableMapping[StrOrBytes, StrOrBytes],
```

Comment:
Should we consider submitting this change as its own "fix headers type annotations" PR? Very similar to #976. (We tend to strongly go for "one PR = one change", mostly because then it's easier to decide what goes into release changelogs and what don't.)

Reply:
Yes, sure.

Reply:
FYI I'm in the process of opening a PR for this that also resolves the odd mypy error you mentioned here https://github.com/encode/httpx/pull/985#discussion_r429624707 :-)

Reply:
Thanks for the PR :+1: . Yes, I agree with you. this change will save our time and we can go faster.